lib/utils.pm

```diff
@@ -190,6 +191,16 @@ sub is_casp {
     return check_var('FLAVOR', $flavor);
 }
 
+# Checks if hypervisor is xen with optional type [pv|hvm] parameter
+sub is_xen {
+    my $type = shift;
+    return 0 unless check_var('VIRSH_VMM_FAMILY', 'xen');
+    return 1 unless $type;
+
+    return check_var('VIRSH_VMM_TYPE', 'linux') if $type eq 'pv';
+    return check_var('VIRSH_VMM_TYPE', $type);
+}
```

Comment:
I think this function over complicated exactly because you try to use two variables ('linux' and 'pv') for same thing XEN=PV.  Don't think it is good idea it will confuse people a lot. I think need to select single one and use it everywhere. 
In this case you will be able drop this function and just use check_var('VIRSH_VMM_TYPE','...... ')  in all place where you currently calling is_xen